### ISSUE TITLE
Installation issue: sample hip-tests have segfault when compiled with hip@5.6.0 (or other)

### ISSUE THREAD

**Issue by dguibert** [LLNL/spack#40560]
### Steps to reproduce the issue

```console
$ spack spec -I hip@5.6.0
Input spec
--------------------------------
 -   hip@5.6.0 arch=None-None-x86_64

Concretized
--------------------------------
[+]  zuvpwe6  hip@5.6.0%gcc@8.5.0~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=aee7249,c2ee21c,e73e91b arch=linux-rhel8-x86_64
[+]  wgxjsla      ^cmake@3.27.6%gcc@8.5.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-rhel8-x86_64
[e]  2gukc4e          ^curl@7.61.1%gcc@8.5.0+gssapi+ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-rhel8-x86_64
[e]  bj65xtx          ^ncurses@6.1.20180224%gcc@8.5.0~symlinks+termlib abi=6 build_system=autotools arch=linux-rhel8-x86_64
[+]  pwvjwcy          ^zlib@1.2.13%gcc@8.5.0+optimize+pic+shared build_system=makefile arch=linux-rhel8-x86_64
[+]  5ltvzkr      ^comgr@5.6.0%gcc@8.5.0~ipo build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[+]  vres4cc          ^rocm-cmake@5.6.0%gcc@8.5.0~ipo build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[+]  2gmoqbs          ^z3@4.11.2%gcc@8.5.0~gmp~ipo~python build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[+]  t26ukiu      ^glx@1.4%gcc@8.5.0 build_system=bundle arch=linux-rhel8-x86_64
[+]  loaycsd          ^mesa@23.0.3%gcc@8.5.0+glx+llvm+opengl~opengles+osmesa~strip build_system=meson buildtype=release default_library=shared arch=linux-rhel8-x86_64
[+]  4opsrt2              ^bison@3.8.2%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  p426j3k                  ^diffutils@3.9%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  svpgb4l              ^expat@2.5.0%gcc@8.5.0+libbsd build_system=autotools arch=linux-rhel8-x86_64
[+]  yxwycre                  ^libbsd@0.11.7%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  p2hdwva                      ^libmd@1.0.4%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  wddh3gs              ^flex@2.6.3%gcc@8.5.0+lex~nls build_system=autotools arch=linux-rhel8-x86_64
[+]  byxbwkz                  ^findutils@4.9.0%gcc@8.5.0 build_system=autotools patches=440b954 arch=linux-rhel8-x86_64
[+]  chfprod              ^gettext@0.21.1%gcc@8.5.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-rhel8-x86_64
[+]  q7lww4u                  ^libxml2@2.10.3%gcc@8.5.0~python build_system=autotools arch=linux-rhel8-x86_64
[e]  ljzger4                  ^tar@1.30%gcc@8.5.0 build_system=autotools zip=pigz arch=linux-rhel8-x86_64
[+]  5ey2fzw              ^glproto@1.4.17%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  3qnwm34                  ^util-macros@1.19.3%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  b5h7jbd              ^libunwind@1.6.2%gcc@8.5.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs~pic+tests+weak_backtrace~xz~zlib build_system=autotools components=none libs=shared,static arch=linux-rhel8-x86_64
[+]  jxguwib              ^libx11@1.8.4%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  d47jypr                  ^inputproto@2.3.2%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  x6hqazr                  ^kbproto@1.0.7%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  c3wcy6n                  ^xextproto@7.3.0%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  5t5bzak                  ^xproto@7.0.31%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  om3am7k                  ^xtrans@1.4.0%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  6pd6xkh              ^libxcb@1.14%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  hm2k7an                  ^libxau@1.0.8%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  mrxumt6                  ^libxdmcp@1.1.4%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  psxb7st                  ^xcb-proto@1.15.2%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  zlj4wor              ^libxext@1.3.3%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  py2s6ga              ^libxt@1.1.5%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  2jkqmug                  ^libice@1.0.9%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  mq6moir                  ^libsm@1.2.3%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  cth67kz              ^llvm@16.0.6%gcc@8.5.0+clang~cuda~flang+gold~ipo+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+polly~python~split_dwarf~z3~zstd build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=runtime shlib_symbol_version=none targets=all version_suffix=none arch=linux-rhel8-x86_64
[+]  hinqgie                  ^binutils@2.40%gcc@8.5.0~gas+gold~gprofng+headers~interwork+ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs=shared,static arch=linux-rhel8-x86_64
[e]  qevmeqx                      ^diffutils@3.6%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  pgf5m3d                      ^texinfo@7.0.3%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  cx7ub4x                  ^hwloc@2.9.1%gcc@8.5.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rhel8-x86_64
[+]  nmjlyxl                  ^libedit@3.1-20210216%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  ukf4rfg                  ^lua@5.3.6%gcc@8.5.0~pcfile+shared build_system=makefile fetcher=curl arch=linux-rhel8-x86_64
[+]  e5hx2ai                      ^unzip@6.0%gcc@8.5.0 build_system=makefile arch=linux-rhel8-x86_64
[+]  6lymkvt                  ^perl-data-dumper@2.173%gcc@8.5.0 build_system=perl arch=linux-rhel8-x86_64
[+]  cy2h5to                  ^swig@4.1.1%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  firt3ll                      ^pcre2@10.42%gcc@8.5.0~jit+multibyte build_system=autotools arch=linux-rhel8-x86_64
[+]  23yvhy3              ^meson@1.2.1%gcc@8.5.0 build_system=python_pip patches=0f0b1bd arch=linux-rhel8-x86_64
[+]  cmoctcf              ^py-mako@1.2.4%gcc@8.5.0 build_system=python_pip arch=linux-rhel8-x86_64
[+]  wx2lu2b                  ^py-markupsafe@2.1.3%gcc@8.5.0 build_system=python_pip arch=linux-rhel8-x86_64
[+]  c5bozsj              ^xrandr@1.5.0%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  bqgj25f                  ^libxrandr@1.5.3%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  jqgya7h                      ^renderproto@0.11.1%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  j73dpsh                  ^libxrender@0.9.10%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  lftvilg                  ^randrproto@1.5.0%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  mmzg3gy      ^gmake@4.4.1%gcc@8.5.0~guile build_system=autotools arch=linux-rhel8-x86_64
[+]  vvi7e7k      ^hipify-clang@5.6.0%gcc@8.5.0~ipo build_system=cmake build_type=Release generator=make patches=54b8b39 arch=linux-rhel8-x86_64
[+]  dluw75q      ^hsa-rocr-dev@5.6.0%gcc@8.5.0+image~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[+]  oz73fjd          ^elfutils@0.189%gcc@8.5.0~debuginfod+exeprefix+nls build_system=autotools arch=linux-rhel8-x86_64
[+]  kov2kr2              ^libiconv@1.17%gcc@8.5.0 build_system=autotools libs=shared,static arch=linux-rhel8-x86_64
[+]  jpnp5gc              ^m4@1.4.19%gcc@8.5.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rhel8-x86_64
[+]  atlcywf                  ^libsigsegv@2.14%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[e]  x6krjkd              ^xz@5.2.4%gcc@8.5.0~pic build_system=autotools libs=shared,static arch=linux-rhel8-x86_64
[+]  4g6n4sz              ^zstd@1.5.5%gcc@8.5.0~programs build_system=makefile libs=shared,static arch=linux-rhel8-x86_64
[+]  tliyldg          ^pkgconf@1.9.5%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  i7bfowh          ^xxd-standalone@8.2.1201%gcc@8.5.0 build_system=makefile arch=linux-rhel8-x86_64
[+]  u7rxwrw      ^hsakmt-roct@5.6.0%gcc@8.5.0~ipo+shared build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[+]  auckvpn          ^libdrm@2.4.115%gcc@8.5.0~docs build_system=generic arch=linux-rhel8-x86_64
[+]  2azqf72              ^libpciaccess@0.17%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  uwfohvf                  ^libtool@2.4.7%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[e]  7etxjng                      ^m4@1.4.18%gcc@8.5.0+sigsegv build_system=autotools patches=3877ab5,fc9b616 arch=linux-rhel8-x86_64
[-]  pn7oh53                  ^util-macros@1.19.3%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  wgsijbs              ^libpthread-stubs@0.4%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  xz4fyxh      ^llvm-amdgpu@5.6.0%gcc@8.5.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_system=cmake build_type=Release generator=ninja patches=a08bbe1,c4750bb,d35aec9 arch=linux-rhel8-x86_64
[+]  6uybcmz          ^ninja@1.11.1%gcc@8.5.0+re2c build_system=generic arch=linux-rhel8-x86_64
[+]  q6kqts7              ^re2c@2.2%gcc@8.5.0 build_system=generic arch=linux-rhel8-x86_64
[+]  ysjrz6u          ^python@3.9.16%gcc@8.5.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,4c24573,f2fd060 arch=linux-rhel8-x86_64
[+]  xuwci3j              ^libffi@3.4.4%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  bsqa3jt              ^libxcrypt@4.4.33%gcc@8.5.0~obsolete_api build_system=autotools arch=linux-rhel8-x86_64
[+]  rslydu4              ^openssl@1.1.1t%gcc@8.5.0~docs~shared build_system=generic certs=mozilla arch=linux-rhel8-x86_64
[-]  l7frxqr                  ^ca-certificates-mozilla@2023-01-10%gcc@8.5.0 build_system=generic arch=linux-rhel8-x86_64
[e]  qbpp2be              ^pkgconf@1.4.2%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  hwqr4y5              ^readline@8.2%gcc@8.5.0 build_system=autotools patches=bbf97f1 arch=linux-rhel8-x86_64
[+]  yropwiu              ^sqlite@3.40.1%gcc@8.5.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-rhel8-x86_64
[+]  2bpcegj              ^util-linux-uuid@2.38.1%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  6qvzxlv      ^numactl@2.0.14%gcc@8.5.0 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-rhel8-x86_64
[e]  4lhmbak          ^autoconf@2.69%gcc@8.5.0 build_system=autotools patches=7793209 arch=linux-rhel8-x86_64
[e]  3ucwd4a          ^automake@1.16.1%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  occqzhd          ^libtool@2.4.7%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[e]  5pbnh6e          ^m4@1.4.18%gcc@8.5.0+sigsegv build_system=autotools patches=3877ab5,fc9b616 arch=linux-rhel8-x86_64
[+]  iyhitrm      ^perl@5.36.0%gcc@8.5.0+cpanm+open+shared+threads build_system=generic arch=linux-rhel8-x86_64
[+]  354jm66          ^berkeley-db@18.1.40%gcc@8.5.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-rhel8-x86_64
[e]  nzqyho7          ^bzip2@1.0.6%gcc@8.5.0~debug~pic+shared build_system=generic arch=linux-rhel8-x86_64
[+]  xybclvx          ^gdbm@1.23%gcc@8.5.0 build_system=autotools arch=linux-rhel8-x86_64
[+]  52bzv7v      ^perl-file-which@1.27%gcc@8.5.0 build_system=perl arch=linux-rhel8-x86_64
[+]  mtoelg3      ^perl-uri-encode@1.1.1%gcc@8.5.0 build_system=perl arch=linux-rhel8-x86_64
[+]  ighmxo5          ^perl-module-build@0.4232%gcc@8.5.0 build_system=perl arch=linux-rhel8-x86_64
[+]  yczxmzo      ^py-cppheaderparser@2.7.4%gcc@8.5.0 build_system=python_pip arch=linux-rhel8-x86_64
[+]  lyr4hmf          ^py-pip@23.1.2%gcc@8.5.0 build_system=generic arch=linux-rhel8-x86_64
[+]  3xxctri          ^py-ply@3.11%gcc@8.5.0 build_system=python_pip arch=linux-rhel8-x86_64
[+]  xomzb3l          ^py-setuptools@63.4.3%gcc@8.5.0 build_system=generic arch=linux-rhel8-x86_64
[-]  xgit5ed              ^py-pip@23.0%gcc@8.5.0 build_system=generic arch=linux-rhel8-x86_64
[+]  tf5v6ny          ^py-wheel@0.37.1%gcc@8.5.0 build_system=generic arch=linux-rhel8-x86_64
[+]  d2othd6      ^rocm-core@5.6.0%gcc@8.5.0~ipo build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[+]  hs5exei      ^rocminfo@5.6.0%gcc@8.5.0~ipo build_system=cmake build_type=Release generator=make arch=linux-rhel8-x86_64
[+]  2kxm2kx      ^roctracer-dev-api@5.6.0%gcc@8.5.0 build_system=generic arch=linux-rhel8-x86_64
```

It compile fine. Then try to compile and run a simple hip test (like square.cu)
````sh
git clone https://github.com/ROCm-Developer-Tools/hip-tests/
cd samples/0_Intro/square
hipify-perl square.cu > square.cpp
hipcc square.cpp -o square
AMD_LOG_LEVEL=4 ./square.spack 
[...]
Segmentation fault (core dumped)
````

@haampie, @renjithravindrankannath, @srekolam any hints?

### Error message

<details><summary>Error message</summary>

<pre>
:3:rocdevice.cpp            :434 : 1141372672719 us: 505623: [tid:0x7f01e4e455c0] Initializing HSA stack.
:3:rocdevice.cpp            :202 : 1141372751517 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[2]=0x560ed2759ab0(fine=0x560ed2759cd0,coarse=0x560ed275a450) for gpu agent=0x560ed28270e0
:3:rocdevice.cpp            :1635: 1141372752132 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372752389 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d6200000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372752750 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d6000000, size 0x101000
:3:rocdevice.cpp            :202 : 1141372752848 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[2]=0x560ed2759ab0(fine=0x560ed2759cd0,coarse=0x560ed275a450) for gpu agent=0x560ed282bc10
:3:rocdevice.cpp            :1635: 1141372753146 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372753389 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5e00000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372753690 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5c00000, size 0x101000
:3:rocdevice.cpp            :202 : 1141372753780 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[2]=0x560ed2759ab0(fine=0x560ed2759cd0,coarse=0x560ed275a450) for gpu agent=0x560ed2830080
:3:rocdevice.cpp            :1635: 1141372754048 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372754262 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5a00000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372754573 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5800000, size 0x101000
:3:rocdevice.cpp            :202 : 1141372754759 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[2]=0x560ed2759ab0(fine=0x560ed2759cd0,coarse=0x560ed275a450) for gpu agent=0x560ed28344f0
:3:rocdevice.cpp            :1635: 1141372755023 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372755225 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5600000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372755556 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5400000, size 0x101000
:3:rocdevice.cpp            :202 : 1141372755744 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[6]=0x560ed280a560(fine=0x560ed280a750,coarse=0x560ed280aed0) for gpu agent=0x560ed2838930
:3:rocdevice.cpp            :1635: 1141372756004 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372756311 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5200000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372756663 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d5000000, size 0x101000
:3:rocdevice.cpp            :202 : 1141372756751 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[6]=0x560ed280a560(fine=0x560ed280a750,coarse=0x560ed280aed0) for gpu agent=0x560ed283cd40
:3:rocdevice.cpp            :1635: 1141372757014 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372757306 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d4e00000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372757607 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d4c00000, size 0x101000
:3:rocdevice.cpp            :202 : 1141372757767 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[6]=0x560ed280a560(fine=0x560ed280a750,coarse=0x560ed280aed0) for gpu agent=0x560ed2841180
:3:rocdevice.cpp            :1635: 1141372758031 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372758258 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d4a00000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372758622 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d4800000, size 0x101000
:3:rocdevice.cpp            :202 : 1141372758811 us: 505623: [tid:0x7f01e4e455c0] Numa selects cpu agent[6]=0x560ed280a560(fine=0x560ed280a750,coarse=0x560ed280aed0) for gpu agent=0x560ed2845590
:3:rocdevice.cpp            :1635: 1141372759076 us: 505623: [tid:0x7f01e4e455c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :2012: 1141372759114 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f01d6d02000, size 0x1c0
:4:rocdevice.cpp            :2012: 1141372759367 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d4600000, size 0x101000
:4:rocdevice.cpp            :2012: 1141372759720 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7f00d4400000, size 0x101000
:4:runtime.cpp              :83  : 1141372759894 us: 505623: [tid:0x7f01e4e455c0] init
:3:hip_context.cpp          :48  : 1141372759899 us: 505623: [tid:0x7f01e4e455c0] Direct Dispatch: 1
:3:hip_device.cpp           :381 : 1141372759947 us: 505623: [tid:0x7f01e4e455c0] hipGetDeviceProperties: Returned hipSuccess : 
info: running on device AMD Instinct MI250X / MI250
info: allocate host mem (  7.63 MB)
info: allocate device mem (  7.63 MB)
:3:hip_memory.cpp           :562 : 1141372761932 us: 505623: [tid:0x7f01e4e455c0]  hipMalloc ( 0x7fff71932e80, 4000000 ) 
:4:rocdevice.cpp            :2141: 1141372761983 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa device memory 0x7ef9cea00000, size 0x3d0900
:3:rocdevice.cpp            :2180: 1141372761990 us: 505623: [tid:0x7f01e4e455c0] device=0x560ed284d920, freeMem_ = 0xffec2f700
:3:hip_memory.cpp           :564 : 1141372761995 us: 505623: [tid:0x7f01e4e455c0] hipMalloc: Returned hipSuccess : 0x7ef9cea00000: duration: 63 us
:3:hip_memory.cpp           :562 : 1141372762006 us: 505623: [tid:0x7f01e4e455c0]  hipMalloc ( 0x7fff71932e78, 4000000 ) 
:4:rocdevice.cpp            :2141: 1141372762030 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa device memory 0x7ef9ce400000, size 0x3d0900
:3:rocdevice.cpp            :2180: 1141372762035 us: 505623: [tid:0x7f01e4e455c0] device=0x560ed284d920, freeMem_ = 0xffe85ee00
:3:hip_memory.cpp           :564 : 1141372762039 us: 505623: [tid:0x7f01e4e455c0] hipMalloc: Returned hipSuccess : 0x7ef9ce400000: duration: 33 us
info: copy Host2Device
:3:hip_memory.cpp           :637 : 1141372762055 us: 505623: [tid:0x7f01e4e455c0]  hipMemcpy ( 0x7ef9cea00000, 0x7f00d402f010, 4000000, hipMemcpyHostToDevice ) 
:3:rocdevice.cpp            :2822: 1141372762063 us: 505623: [tid:0x7f01e4e455c0] number of allocated hardware queues with low priority: 0, with normal priority: 0, with high priority: 0, maximum per priority is: 4
:3:rocdevice.cpp            :2897: 1141372772060 us: 505623: [tid:0x7f01e4e455c0] created hardware queue 0x7f01d6cf2000 with size 16384 with priority 1, cooperative: 0
:3:rocdevice.cpp            :2964: 1141372772071 us: 505623: [tid:0x7f01e4e455c0] acquireQueue refCount: 0x7f01d6cf2000 (1)

:4:rocdevice.cpp            :2012: 1141372772265 us: 505623: [tid:0x7f01e4e455c0] Allocate hsa host memory 0x7ef9ce000000, size 0x100000
:1:devprogram.cpp           :1874: 1141372826569 us: 505623: [tid:0x7f01e4e455c0] ERROR: linking module flags 'amdgpu_code_object_version': IDs have conflicting values in '' and 'llvm-link'
Error: Linking bitcode failed: linking source & IR libraries.

:1:rocdevice.cpp            :859 : 1141372826589 us: 505623: [tid:0x7f01e4e455c0] Couldn't create blit kernels!
:1:rocvirtual.cpp           :1275: 1141372826592 us: 505623: [tid:0x7f01e4e455c0] Could not create BlitManager!
:4:rocdevice.cpp            :2157: 1141372826839 us: 505623: [tid:0x7f01e4e455c0] Free hsa memory 0x7ef9ce000000
:3:rocdevice.cpp            :2976: 1141372826846 us: 505623: [tid:0x7f01e4e455c0] releaseQueue refCount:0x7f01d6cf2000 (0)

:4:command.cpp              :349 : 1141372826859 us: 505623: [tid:0x7f01e4e455c0] Command (CopyHostToDevice) enqueued: 0x560ed28c05d0
Segmentation fault (core dumped)
</pre></details>

### Information on your system

* **Spack:** 0.21.0.dev0 (5847363c92b6132b3f124f405f1a774dfc6a18f2)
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen3
* **Concretizer:** clingo

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by srekolam** [comment]
this will be resolved by #40367

**Comment by dguibert** [comment]
Hi @srekolam,

Thank you, indeed #40367 solves this issue as you mention.
